### PR TITLE
FOGL-2641 - avahi dbus name collision handling issue, does not allow more than 1 FogLAMP instances discoverable over the network 

### PR DESCRIPTION
**Pull request by amarendra-dianomic**

### CODE REVIEW

File: python/foglamp/services/core/server.py [hunk 1]
```diff
@@ -732,10 +732,11 @@ def _start_core(cls, loop=None):
...
             # All services are up so now we can advertise the Admin and User REST API's
-            cls.admin_announcer = ServiceAnnouncer(cls._service_name, cls._ADMIN_API_SERVICE, service_server_port,
-                                                   [cls._service_description])
-            cls.user_announcer = ServiceAnnouncer(cls._service_name, cls._USER_API_SERVICE, service_server_port,
-                                                  [cls._service_description])
+            # cls.admin_announcer = ServiceAnnouncer(cls._service_name, cls._ADMIN_API_SERVICE, service_server_port,
+            #                                        [cls._service_description])
+            # cls.user_announcer = ServiceAnnouncer(cls._service_name, cls._USER_API_SERVICE, service_server_port,
+            #                                       [cls._service_description])
+
```
Comment (MarkRiddoch) [inline python/foglamp/services/core/server.py]: Why are these being removed?

File: python/foglamp/services/common/service_announcer.py [hunk 2]
```diff
@@ -6,52 +6,61 @@
...
 
-    _group = None
-    """ The Avahi group """
+class ServiceAnnouncer:
+    def __init__(self, sname, stype, port, txt):
+        host_name = socket.gethostname()
+        host = self.get_ip()
+        service_name = "_" + sname.lower() + "._tcp.local."
+        desc = {'path': '/~paulsm/'}  # TODO: Change
+        _LOGGER.error(">>>>>>>>>>>>>>>>> %s", service_name)
```
Comment (MarkRiddoch) [inline python/foglamp/services/common/service_announcer.py]: Please do not checkin with this in place.

Comment (amarendra-dianomic) [reply python/foglamp/services/common/service_announcer.py]: Fixed

File: python/foglamp/services/core/server.py (hunk 1 above)
Comment (amarendra-dianomic) [reply python/foglamp/services/core/server.py]: Fixed

**Comment by amarendra-dianomic** [comment]
@MarkRiddoch, @praveen-garg feedback changes done. Please check.

File: python/foglamp/services/common/service_announcer.py [hunk 3]
```diff
@@ -6,52 +6,60 @@
...
-    _service_name = None
-    """ The name of the service to advertise """
 
-    _group = None
-    """ The Avahi group """
+class ServiceAnnouncer:
+    def __init__(self, sname, stype, port, txt):
+        host_name = socket.gethostname()
+        host = self.get_ip()
+        service_name = "{}_{}.{}".format(host_name, sname, stype)
```
Comment (praveen-garg) [inline python/foglamp/services/common/service_announcer.py]: can we simpy keep it the name, as passed? hostname we are already putting in for `server`

**Comment by MarkRiddoch** [comment]
Looks like a good way forward.

**Pull request merged**